File: 1282_group_the_people_given/Java/1282.java
import java.util.*;

class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> answer = new ArrayList<>();
        List<List<Integer>> groups = new ArrayList<>();

        for (int i = 0; i < groupSizes.length; ++i) {
            int size = groupSizes[i];

            int j;
            for (j = 0; j < groups.size(); ++j) {
                if (groups.get(j).get(0) == size && groups.get(j).get(1) > 0) {
                    groups.get(j).add(i);
                    groups.get(j).set(1, groups.get(j).get(1) - 1);
                    break;
                }
            }

            if (j == groups.size()) {
                List<Integer> newGroup = new ArrayList<>();

                newGroup.add(size);
                newGroup.add(size - 1);
                newGroup.add(i);

                groups.add(newGroup);
            }
        }

        for (int i = 0; i < groups.size(); ++i) {
            List<Integer> group = new ArrayList<>();

            for (int j = 2; j < groups.get(i).size(); ++j) {
                group.add(groups.get(i).get(j));
            }

            answer.add(group);
        }

        return answer;
    }
}

class P1282 {
    public static void main(String args[]) {
        int[] groupSizes = {3,3,3,3,3,1,3};

        Solution solution = new Solution();

        System.out.println(solution.groupThePeople(groupSizes));
    }
}

File: 1282_group_the_people_given/C++/1282.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Printer {
public:
    void printAnswer(int answer) {
        cout << answer << '\n';
    }

    void printAnswer(bool answer) {
        cout << (answer ? "true" : "false") << '\n';
    }

    void printAnswer(string answer) {
        cout << answer << '\n';
    }

    void printAnswer(vector<int>& answer) {
        for (int i = 0; i < answer.size(); ++i)
            cout << answer[i] << ' ';
        cout << '\n';
    }

    void printAnswer(vector<vector<int>>& answer) {
        for (int i = 0; i < answer.size(); ++i) {
            for (int j = 0; j < answer[i].size(); ++j)
                cout << answer[i][j] << ' ';
            cout << '\n';
        }
    }
};

class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes){
        vector<vector<int>> groups, answer;

        for (int i = 0; i < groupSizes.size(); ++i) {
            int size = groupSizes[i];

            int j;
            for (j = 0; j < groups.size(); ++j) {
                if (groups[j][0] == size) {
                    if (groups[j][1] > 0) {
                        groups[j].push_back(i);
                        --groups[j][1];
                        break;
                    }
                }
            }

            if (j == groups.size()) {
                vector<int> newGroup {size, size - 1, i};

                groups.push_back(newGroup);
            }
        }

        for (int i = 0; i < groups.size(); ++i) {
            vector<int> group(groups[i].begin() + 2, groups[i].end());

            answer.push_back(group);
        }

        return answer;
    }
};

int main() {
    vector<int> groupSizes {3,3,3,3,3,1,3};
    
    Solution solution;
    Printer printer;

    printer.printAnswer(solution.groupThePeople(groupSizes));

    return 0;
}